src/wayland/buffer.rs

```diff
@@ -150,6 +167,14 @@ impl WaylandBuffer {
     pub fn released(&self) -> bool {
         self.released.load(Ordering::SeqCst)
     }
+
+    fn len(&self) -> usize {
+        self.width as usize * self.height as usize
+    }
+
+    pub unsafe fn mapped_mut(&mut self) -> &mut [u32] {
+        unsafe { slice::from_raw_parts_mut(self.map.as_mut_ptr() as *mut u32, self.len()) }
```

Comment:
MmapMut dereferences to `&mut [u8]`. Is there is particular reason we create the slice ourselves from the pointer?

Reply:
We need an `&mut [u32]`, rather than `&mut [u8]`. So it needs an unsafe cast or transmute either way. We also want to restrict the size to the size of the buffer and not the total capacity of the pool.

So I don't think this is worse than other ways of doing the same thing, at least.

Reply:
Ah okay, well this may change if we want more formats in the future.

Reply:
Yeah, if we want to support formats that aren't 32-bit per pixel, we'll need to figure out what API to expose for that.

Reply:
Just a heads up: I already have `bytemuck` in the dependency tree, so you could probably use `bytemuck::cast_slice_mut` to avoid this unsafe.

Reply:
Technically that adds overhead since `bytemuck` has to check alignment and panic if it's wrong. (And we know the memory is page aligned). Though the added cost is probably irrelevant.

This call is somewhat unsafe as implemented anyway though, since we're getting a mutable reference into shared memory, without checking that the buffer is released. Though I guess if we trust the server not to write to it anyway, any unsafely of mutating a buffer the server is reading isn't on our end...